pkg/cloud-provider/cloudapi/aws/aws_ec2.go

```diff
@@ -45,6 +46,7 @@ type ec2ServiceConfig struct {
 // ec2ResourcesCacheSnapshot holds the results from querying for all instances.
 type ec2ResourcesCacheSnapshot struct {
 	instances   map[cloudcommon.InstanceID]*ec2.Instance
+	vpcs        []*ec2.Vpc
```

Comment:
why we have vpcs and vpcIDs stored separately ??

Reply:
vpcIDs are vpc ids of only the instances that we import. vpcs are added by us in this PR which contains all vpcs in the region. As they both are different I had to retain vpcIDs.

Reply:
We should combine them. There should be 1 storage for VPC. If you need IDs, fetch it from 1 Table..